addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -642,7 +642,7 @@ var SnippetEditor = Widget.extend({
         // Hide the snippetEditor if none of its options are visible
         // This cannot be done using the visibility of the options' UI
         // because some options can be located in the overlay.
-        const $visibleOptions = this.$optionsSection.find('we-top-button-group, we-customizeblock-option')
+        const $visibleOptions = this.$optionsSection.find('we-customizeblock-option')
                 .children(':not(.d-none)');
```

Comment:
I feel like this is not right yet:
1. If the option widget uses `_computeVisibility` and not `_computeWidgetVisibility`, does this work? Since this seems to only consider the children of options (= widgets)?
2. In the end, could not we make something cleaner about this: instead of checking for d-none, have the `updateUIVisibility` method's Promise resolve to a boolean which indicates if any option (and itself) is visible. Then here it would only be about checking any of those Promise resolve to true. Since those promises currently do not resolve to any value, we can change that in stable I'd say.
3. This code seems to have been introduced in 16.0, so maybe it should be fixed over there? https://github.com/odoo/odoo/commit/39d073ccafddf4b3fbe9ec9f5f11849d77c9bdaa
4. I am not sure considering options only is right actually, seems weird: "There are some options for an element, if it has the class 'XXX', there is no visible option though -> would not that be weird that this means that if 'XXX', you cannot remove the element but if there is no 'XXX', you can remove the element? Of course, this is what is currently fixing your issue here but: your case with the media list may actually be a bad example of "an option that may or may not be visible. The media list inner column can *never* become grid items. Maybe the selector for the grid padding option should consider that (maybe considering the `s_col_no_resize`? not sure).

=> So maybe two commits are needed:
- One in 16.0 to fix/imp the visibility stuff and togglingOverlay stuff (point 1, 2, 3 above)
- One here in 17.0 to improve the selector for the grid padding option (point 4 above)